### DISCUSSION TITLE
How to handle Mongoose Cannot populate path is not in your schema error?

### DISCUSSION THREAD

**Discussion by samislam** [Automattic/mongoose#12182] [Q&A]
> MongooseError: Cannot populate path `friends` because it is not in your
> schema. Set the `strictPopulate` option to false to override.

I have the following code:
```js
const userSchema = mongoose.Schema({
   age: Number
})
const UserModel = mongoose.model('User', userSchema, 'users')
```

and then in one of the controllers I have the following code:
```js
UserModel.find({ name: 'Isaa' }).populate('friends'); 
```

The above line results in the following error:

> MongooseError: Cannot populate path `friends` because it is not in your
> schema. Set the `strictPopulate` option to false to override.

I want to handle this error in my express error handling middleware, but how?
The error doesn't have any unique field that I can use, such as `error.code` or `error.name`.
So How to handle this error?

```js
app.use((err, req, res, next) => {
    // 👈 here
})
```

**Comment by vkarpov15** [comment]
We should add a separate error class for this error. Right now there is no good way to handle just this error, other than relying on the error message.
